### PR TITLE
Fix issue 592

### PR DESCRIPTION
**Pull request by ggoranov**
How to test: `go test github.com/dusk-network/dusk-blockchain/pkg/core/chain/... -v -race -run TestConcurrentBlock`

### CODE REVIEW

File: harness/tests/localnet_test.go [hunk 1]
```diff
@@ -107,7 +107,14 @@ func TestSendBidTransaction(t *testing.T) {
 	t.Log("Send request to node 0 to generate and process a Bid transaction")
 	txidBytes, err := localNet.SendBidCmd(0, 10, 10)
 	if err != nil {
-		t.Fatal(err.Error())
+		t.Fatal(err)
+	}
+
+	for i := 1; i < localNet.Size(); i++ {
```
Comment (devtomnl) [inline harness/tests/localnet_test.go:113]: so we are now going to always test with sending bids to all participants by default ?

File: pkg/core/chain/chain.go [hunk 2]
```diff
@@ -200,17 +202,26 @@ func New(ctx context.Context, eventBus *eventbus.EventBus, rpcBus *rpcbus.RPCBus
...
 	return chain, nil
 }
 
 // Listen to the collectors
 func (c *Chain) Listen() {
 	for {
 		select {
+		case m := <-c.blockChan:
+			if err := c.onAcceptBlock(m); err != nil {
+				log.WithError(err).Warnf("Handling %s failed", topics.Block.String())
```
Comment (devtomnl) [inline pkg/core/chain/chain.go:219]: could you please log it without Warnf and split the fields? that helps with the monitoring.

File: pkg/core/chain/chain.go [hunk 3]
```diff
@@ -200,17 +202,26 @@ func New(ctx context.Context, eventBus *eventbus.EventBus, rpcBus *rpcbus.RPCBus
...
+			}
+		case m := <-c.initializationChan:
+			if err := c.onInitialization(m); err != nil {
+				log.WithError(err).Warnf("Handling %s failed", topics.Initialization.String())
```
Comment (devtomnl) [inline pkg/core/chain/chain.go:223]: same here

File: pkg/util/nativeutils/eventbus/listenermap.go [hunk 4]
```diff
@@ -36,8 +36,10 @@ func (h *listenerMap) Store(key topics.Topic, value Listener) uint32 {
 func (h *listenerMap) Load(key topics.Topic) []idListener {
 	h.lock.RLock()
 	listeners := h.listeners[key]
+	dup := make([]idListener, len(listeners))
+	copy(dup, listeners)
 	h.lock.RUnlock()
```
Comment (devtomnl) [inline pkg/util/nativeutils/eventbus/listenermap.go:41]: could we add the unlock with defer on the begin of the func ? just for consistency.

File: harness/tests/localnet_test.go (hunk 1 above)
Comment (ggoranov) [reply harness/tests/localnet_test.go:113]: Yes, that's the idea. Such situation is closer to what you can expect in mainnet (multiple bidders network)

File: pkg/util/nativeutils/eventbus/listenermap.go (hunk 4 above)
Comment (ggoranov) [reply pkg/util/nativeutils/eventbus/listenermap.go:41]: No, defer is not needed here.

File: pkg/core/chain/chain.go (hunk 2 above)
Comment (ggoranov) [reply pkg/core/chain/chain.go:219]: No, this still can be easily parsed by monitoring.

**Comment by devtomnl** [comment]
LGTM, all tests are passing.

**Pull request merged**